accepted/future-releases/primary-constructors/feature-specification.md

```diff
@@ -782,13 +805,13 @@ Let `p` be a formal parameter in _k_ which has the modifier `var` or the
 modifier `final` *(that is, `p` is a declaring parameter)*.
 
 Consider the situation where `p` has no type annotation:
-- if combined member signature for a getter with the same name as `p` from
-  the superinterfaces of _D_ exists and has return type `T`, the parameter
-  `p` has declared type `T`. If no such getter exists, but a setter with
-  the same basename exists, with a formal parameter whose type is `T`, the
-  parameter `p` has declared type `T`. *In other words, an instance
-  variable introduced by a declaring parameter is subject to override
-  inference, just like an explicitly declared instance variable.*
+- if the combined member signature for a getter with the same name as `p`
+  from the superinterfaces of _D_ exists and has return type `T`, the
+  parameter `p` has declared type `T`. If no such getter exists, but a
+  setter with the same basename exists, with a formal parameter whose type
+  is `T`, the parameter `p` has declared type `T`. *In other words, an
+  instance variable introduced by a declaring parameter is subject to
+  override inference, just like an explicitly declared instance variable.*
```

Comment:
Is it worth commenting that this works *only* for declaring constructors? One could think that `Foo([final x = 42, y = 37])` would infer `int` for `y` as well.

_We could do that, but then it would not just be for constructor parameters, so it's a larger change. For now it's only for a declaring parameter, because it **looks like a variable declarations with an initializer expression**, and those do get inference._